lib/improver/blending/weighted_blend.py

```diff
@@ -454,165 +454,216 @@ def process(self, cube, weights=None):
                         one value. Returns the original cube in this case.
 
         """
+
+        self._check_input_cube_is_iris_cube(cube)
+
+        # if the coordinate being blended is forecast_reference_time.
+        if self.coord == "forecast_reference_time":
+            self._check_all_time_points_equal(cube)
+
+        if weights is not None:
+            self._check_weights_have_coord_shape(cube, weights)
```

Comment:
The next bit in the old code is `_get_perc_dim_coord` equivalent. Because this check isn't done here, is there a potential problem within the _check_weights method?

Reply:
I don't think so. The only part of `self` that is used in the check weights method is `self.coord`, which is not modified by `get_perc_dim_coord` or anything else.

I just moved the `_get_perc_dim_coord` stuff into one of the internal methods to reduce the number of arguments that needed to be passed around (didn't want to pass in the percentile coordinate to routines that didn't need to know about it).